core/src/main/scala/org/apache/spark/rdd/JdbcRDD.scala

```diff
@@ -64,8 +64,8 @@ class JdbcRDD[T: ClassTag](
     // bounds are inclusive, hence the + 1 here and - 1 on end
     val length = 1 + upperBound - lowerBound
     (0 until numPartitions).map(i => {
-      val start = lowerBound + ((i * length) / numPartitions).toLong
-      val end = lowerBound + (((i + 1) * length) / numPartitions).toLong - 1
+      val start = lowerBound + ((BigDecimal(i) * length) / numPartitions).toLong
```

Comment:
@srowen 

1. BigInt in scala actually works, I should change to that. 
2. The overflow is not necessary to be only the last partition, in my test case it was the last 4 partitions
3. I think the current algo give us better partitions than the fix interval algo, for the reason in above comment, would you agree with that ?
4. The current algo does +1 on the length and -1 on the end of each interval, I think that works fine.